src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -518,6 +518,18 @@ class Manager : public DoutPrefixProvider {
           }
         }
       }
+
+      // updating perfcounters with topic stats
+      uint64_t entries_size;
+      uint32_t entries_number;
+      const auto ret = cls_2pc_queue_get_topic_stats(rados_ioctx, queue_name, entries_number, entries_size);
+      if (ret < 0) {
+        ldpp_dout(this, 1) << "ERROR: topic stats for topic: " << queue_name << ". error: " << ret << dendl;
+      }
+
+      auto counters = rgw::persistent_topic_counters::get(queue_name);
+      rgw::persistent_topic_counters::set(counters, l_rgw_persistent_topic_len, entries_number);
+      rgw::persistent_topic_counters::set(counters, l_rgw_persistent_topic_size, entries_size);
```

Comment:
what does this look like when there are several radosgws running? we're reading global stats from rados, but exporting them as per-radosgw perf counters. other counters just get summed over all radosgws, but trying to sum these will give the wrong result

Reply:
at a specific point in time, only one RGW is managing a queue. so, the value in other RGWs will be zero.

Reply:
we use cls lock to coordinate this, right? it seems like a lock timeout could allow another radosgw to take over processing of a topic/queue, leaving stale counters in the first

mainly just wanted to raise this as a potential dimensioning issue. these daemon perf counters aren't a great fit for global stats